src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -97,260 +103,231 @@ static bool anyInterval(Interval interval, BinaryOperatorKind relation, T value,
-                                case Equal:
-                                    return tc.Related(Equal, minValue, value) && tc.Related(Equal, maxValue, value);
-                                default:
-                                    throw new ArgumentException(nameof(relation));
-                            }
-                        case Interval.Mixed mixed:
-                            switch (relation)
-                            {
-                                case LessThan when tc.Related(LessThanOrEqual, value, minValue):
-                                    return false;
-                                case LessThan when tc.Related(LessThan, maxValue, value):
-                                    return true;
-                                case GreaterThan when tc.Related(GreaterThanOrEqual, value, maxValue):
-                                    return false;
-                                case GreaterThan when tc.Related(GreaterThan, minValue, value):
-                                    return true;
-                                case LessThanOrEqual when tc.Related(LessThan, value, minValue):
-                                    return false;
-                                case LessThanOrEqual when tc.Related(LessThanOrEqual, maxValue, value):
-                                    return true;
-                                case GreaterThanOrEqual when tc.Related(GreaterThan, value, maxValue):
-                                    return false;
-                                case GreaterThanOrEqual when tc.Related(GreaterThanOrEqual, minValue, value):
-                                    return true;
-                                default:
-                                    var (leftMax, rightMin) = tc.Partition(minValue, maxValue);
-                                    return
-                                        allInterval(mixed.Left, relation, value, minValue, leftMax) &&
-                                        allInterval(mixed.Right, relation, value, rightMin, maxValue);
-                            }
-                        default:
-                            throw new ArgumentException(nameof(interval));
-                    };
+                    case LessThan:
+                    case LessThanOrEqual:
+                        return _intervals.Length == 0 || tc.Related(relation, _intervals[_intervals.Length - 1].last, value);
+                    case GreaterThan:
+                    case GreaterThanOrEqual:
+                        return _intervals.Length == 0 || tc.Related(relation, _intervals[0].first, value);
+                    case Equal:
+                        return _intervals.Length == 0 ||
+                            _intervals.Length == 1 && tc.Related(Equal, _intervals[0].first, value) && tc.Related(Equal, _intervals[0].last, value);
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(relation);
                 }
```

Comment:
Consider checking `if (_intervals.Length == 0) return true;` before the `switch` statement.